src/SoftwareSerial.cpp

```diff
@@ -50,8 +50,27 @@ bool SoftwareSerial::isValidGPIOpin(int8_t pin) {
 #if defined(ESP8266)
     return (pin >= 0 && pin <= 16) && !isFlashInterfacePin(pin);
 #elif defined(ESP32)
+
+  #ifdef CONFIG_IDF_TARGET_ESP32
+    // Datasheet https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf,
+    // Pinout    https://docs.espressif.com/projects/esp-idf/en/latest/esp32/_images/esp32-devkitC-v4-pinout.jpg
     return (pin >= 0 && pin <= 5) || (pin >= 12 && pin <= 19) ||
```

Comment:
I have added the missing pins for the original ESP32 using the format that you had used previously. I had left these out so that we did not have to debate impact on existing code.

The flash pins vary across devices, so it is not practical to try and implement a local definition, for example, it has no meaning on the S2 and C3 variants. If the vendor was to produce such a macro, then that could be used. I don't think its for me to second guess the vendor. I assume this is also why Espressif didn't implement such a macro so far on the ESP32 family.